conda/recipes/libcuml/meta.yaml

```diff
@@ -48,28 +47,54 @@ requirements:
     - libraft-distance {{ minor_version }}
     - libraft-nn {{ minor_version }}
     - lapack
-    - treelite=2.3.0
+    - treelite {{ treelite_version }}
     - faiss-proc=*=cuda
-    - gtest=1.10.0
+    - gtest {{ gtest_version }}
     - gmock
-    - libfaiss 1.7.0 *_cuda
-  run:
-    - libcumlprims {{ minor_version }}
-    - libraft-headers {{ minor_version }}
-    - libraft-distance {{ minor_version }}
-    - libraft-nn {{ minor_version }}
-    - cudf {{ minor_version }}
-    - nccl>=2.9.9
-    - ucx-py {{ ucx_py_version }}
-    - ucx-proc=*=gpu
-    - {{ pin_compatible('cudatoolkit', max_pin='x', min_pin='x') }}
-    - treelite=2.3.0
-    - faiss-proc=*=cuda
-    - libfaiss 1.7.0 *_cuda
-    - libcusolver>=11.2.1
+    - libfaiss {{ libfaiss_version }} *_cuda
+
+outputs:
+  - name: libcuml
+    version: {{ version }}
+    script: install_libcuml.sh
+    build:
+      number: {{ GIT_DESCRIBE_NUMBER }}
+      string: cuda{{ cuda_major }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
+    requirements:
+      build:
+        - cmake {{ cmake_version }}
+      run:
+        - cudatoolkit {{ cuda_spec }}
+        - libcumlprims {{ minor_version }}
+        - libraft-headers {{ minor_version }}
+        - libraft-distance {{ minor_version }}
+        - libraft-nn {{ minor_version }}
+        - cudf {{ minor_version }}
+        - nccl {{ nccl_version }}
+        - ucx-py {{ ucx_py_version }}
+        - ucx-proc=*=gpu
+        - treelite {{ treelite_version }}
+        - faiss-proc=*=cuda
+        - libfaiss {{ libfaiss_version }} *_cuda
+        - libcusolver {{ libcusolver_version }}
 
-about:
-  home: http://rapids.ai/
-  license: Apache-2.0
-  # license_file: LICENSE
-  summary: libcuml library
+    about:
+      home: http://rapids.ai/
+      license: Apache-2.0
+      summary: libcuml library
+  - name: libcuml-tests
+    version: {{ version }}
+    script: install_libcuml_tests.sh
+    build:
+      number: {{ GIT_DESCRIBE_NUMBER }}
+      string: cuda{{ cuda_major }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
+    requirements:
+      build:
+        - cmake {{ cmake_version }}
+      run:
+        - cudatoolkit {{ cuda_spec }}
+        - {{ pin_subpackage('libcuml', exact=True) }}
```

Comment:
For the `libcudf` and `librmm` changes, I had to add `gtests` and `gmock` to the runtime dependencies of the `*-tests` package in order for the tests to run correctly. You may run into a similar issue here. I'm wondering if you can also remove `gtests` and `gmock` from the `host` dependencies of the top-level build, but that might be outside of the scope of this PR.

- https://github.com/rapidsai/rmm/blob/7434bd6bd600d17d4b63e347960042bd81daa3e3/conda/recipes/librmm/meta.yaml#L98-L102

Reply:
Done, but I think we need to keep them as build dependencies or it will fail to find the libs while compiling the tests